python_modules/dagster/dagster/_core/definitions/observable_asset.py

```diff
@@ -50,3 +51,36 @@ def an_asset() -> None:
         )
 
     return an_asset
+
+
+def create_unexecutable_observable_assets_def_from_source_asset(source_asset: SourceAsset):
+    check.invariant(
+        source_asset.observe_fn is None,
+        "Observable source assets not supported yet: observe_fn should be None",
+    )
+    check.invariant(
+        source_asset.auto_observe_interval_minutes is None,
+        "Observable source assets not supported yet: auto_observe_interval_minutes should be None",
+    )
+
+    kwargs = {
+        "key": source_asset.key,
+        "metadata": {
+            **source_asset.metadata,
+            **{SYSTEM_METADATA_KEY_ASSET_EXECUTION_TYPE: AssetExecutionType.UNEXECUTABLE.value},
+        },
```

Comment:
well need to ensure `is_source` get toggled on the `ExternalAssetNode` somehow, maybe other metadata or another update to the execution type enum